### PR TITLE
Add preliminary Mojave support for kext

### PR DESCRIPTION
**Pull request by nuche**
@vit9696 thanks for moving this over and welcoming the pr. This is my first PR and I was cautious about not including a bunch of updates (that could be cruft) from Xcode, except for the added `xcshareddata` folder (which maybe I didn't need to do).

Please let me know if I screwed up anywhere and/or if some additional changes are required. I haven't actually read into the code yet, so I don't know if there are more logical updates that need to be made for this to work PROPERLY. I'm happy to help out any way possible to keep this alive.

### CODE REVIEW

File: kext/sysent.c [hunk 1]
```diff
@@ -254,7 +255,7 @@ bruteforce_sysent(mach_vm_address_t *out_kernel_base)
     *out_kernel_base = kernel_base;
     uint64_t segment_address = 0;
     uint64_t segment_size = 0;
-	if (version_major >= SIERRA) {
+	if (version_major >= MOJAVE) {
```
Comment (vit9696) [inline kext/sysent.c:258]: This changes looks wrong to me, it must stay as before for onyx to remain working on 10.12 and 10.13.

**Comment by vit9696** [comment]
I spotted one change that seems to be off and mentioned it inline.

This is a good start, but if ptrace hooking is still not working as you mentioned in bugreport, it may be that the sysent structure got changed again in 10.14. Check g_sysent_yos type against 10.14 XNU source code, in my opinion.

**Comment by nuche** [comment]
Do we only have 10.14.1 xnu available to us: https://opensource.apple.com/release/macos-10141.html ?

**Comment by nuche** [comment]
Thank you @vit9696 for catching that. I didn't look at that closely so much appreciated.

On the `g_sysent_yos` issue: I have the source from above since it seems to be the only xnu source available (xnu-4903.221.2). Are you referring to looking into [`bsd/sys/sysent.h`](https://opensource.apple.com/source/xnu/xnu-4903.221.2/bsd/sys/sysent.h.auto.html)? Or is there an implementation file I'm missing? I'm not entirely following what to hunt for since I'm a bit rusty on my c.

**Comment by vit9696** [comment]
The 10.14.1 source is currently the latest, and seems to be mostly recent. sysent does not seem to be different for it, so I am not fully positive what to check first at the moment.

**Comment by nuche** [comment]
Hmmmm, will try to see if anyone else has some thoughts. Freenode #OSXRE seems like a good place to get some input.

Some remarks by one person about onyx:

> i think that rootfool code finds some offsets dynamically
> so you just need to do the same with sysent and other offsets to remove dependencies
> 

While we're trying to figure this out, maybe it would make sense to also update the project with the recommendations that Xcode is spewing?

**Comment by vit9696** [comment]
I merged the changes and added some fixes including the one I mentioned above and Xcode project update. I also checked sysent types in 10.14 and 10.15 with IDA, and they seem to be just fine.

If any symbols are not located, you will always be warned about it through the debug log. I do not have plans to debug the issues right now, so if you figure out why that particular antiprace trick does not work, it will be nice.

**Pull request merged**